powerauth-push-server/src/main/java/com/wultra/push/service/PushSendingWorker.java

```diff
@@ -488,31 +476,24 @@ void sendMessageToApns(final ApnsClient apnsClient, final PushMessageBody pushMe
             if (response != null) {
                 final UUID apnsId = response.getApnsId();
                 if (response.isAccepted()) {
-                    logger.info("Notification sent successfully, APNs ID: {}.", apnsId);
+                    logger.info("action: sendMessageToApns, state: succeeded, apnsId: {}", apnsId);
                     callback.didFinishSendingMessage(PushSendingCallback.Result.OK);
                 } else {
                     final Optional<String> rejectionReasonOptional = response.getRejectionReason();
                     String rejectionReason = null;
                     if (rejectionReasonOptional.isPresent()) {
                         rejectionReason = rejectionReasonOptional.get();
                     }
-
-                    if (ApnsRejectionReason.EXPIRED_PROVIDER_TOKEN.isEqualToText(rejectionReason)) {
-                        logger.info("Notification rejected by the APNs gateway due to expired push token, APNs ID: {}.", response.getApnsId());
-                    } else {
-                        logger.info("Notification rejected by the APNs gateway: {}, APNs ID: {}.", rejectionReason != null ? rejectionReason : "UnknownReason", response.getApnsId());
-                    }
+                    logger.warn("action: sendMessageToApns, state: failed, apnsId: {}, reason: {}, topic: {}", apnsId, rejectionReason != null ? rejectionReason : "UnknownReason", apnsTopic);
 
                     // Determine if the push token should be deleted.
                     if (ApnsRejectionReason.BAD_DEVICE_TOKEN.isEqualToText(rejectionReason)) {
                         logger.debug("Deleting push token: {}.", pushToken);
                         callback.didFinishSendingMessage(PushSendingCallback.Result.FAILED_DELETE);
                     } else if (ApnsRejectionReason.DEVICE_TOKEN_NOT_FOR_TOPIC.isEqualToText(rejectionReason)) {
-                        logger.warn("Notification was sent to incorrect topic: {}.", apnsTopic);
```

Comment:
👀 Was it really superfluous?

Reply:
All error logs are replaced by the structure above. The data are the same. This is the exact difference between structured and unstructured logging. We do not want to create redundant messages.